### ISSUE TITLE
[TypeScript] TSServer `createFileSystemWatcher` crashes with yarn PnP

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#44097]
TL;DR `"typescript.tsserver.experimental.useVsCodeWatcher": true` + yarn PnP = TSServer constantly crashing. Related Tickets  -210687, -208872 (MCVE: https://github-com/BurningEnlightenment/reprex-vscode-ts-filewatcher)

After starting VSCode today I got greeted with an error about TSServer being unavailable due to repeated crashes without any helpful log output:
```
2024-05-05 09:15:01.395 [info] Starting TS Server
2024-05-05 09:15:01.395 [info] Using tsserver from: d:\devel\source\zenbox\repro-vscode-filewatcher\.yarn\sdks\typescript\lib\tsserver.js
2024-05-05 09:15:01.395 [info] <syntax> Forking...
2024-05-05 09:15:01.395 [info] <syntax> Starting...
2024-05-05 09:15:01.395 [info] <semantic> Forking...
2024-05-05 09:15:01.395 [info] <semantic> Starting...
2024-05-05 09:15:01.693 [error] TSServer exited. Code: 1. Signal: null
```
Enabling the verbose log didn't reveal anything interesting either:
```
Info 0    [08:23:19.874] Starting TS Server
Info 1    [08:23:19.874] Version: 5.4.5
Info 2    [08:23:19.874] Arguments: D:\Program Files\Microsoft VS Code\Code.exe d:\devel\source\zenbox\repro-vscode-filewatcher\.yarn\sdks\typescript\lib\tsserver.js --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName C:\Users\<redacted>\AppData\Local\Temp\vscode-typescript\3ef5e001a09f1e55f983\tscancellation-707e108b83851806c795.tmp* --logVerbosity verbose --logFile c:\Users\<redacted>\AppData\Roaming\Code\logs\20240505T082314\window1\exthost\vscode.typescript-language-features\tsserver-log-yHjYLz\tsserver.log --locale en --noGetErrOnBackgroundUpdate --canUseWatchEvents --validateDefaultNpmLocation --useNodeIpc
Info 3    [08:23:19.874] Platform: win32 NodeVersion: v18.18.2 CaseSensitive: true
Info 4    [08:23:19.875] ServerMode: undefined hasUnknownServerMode: undefined
Info 5    [08:23:19.877] Binding...
Info 6    [08:23:19.885] FileWatcher:: Added:: WatchInfo: d:\devel\source\zenbox\repro-vscode-filewatcher\.pnp.cjs 250 undefined WatchType: Config file
```
So I started flipping extension settings and was able to narrow it down to `typescript.tsserver.experimental.useVsCodeWatcher`. If enabled `TSServer` started crashing and if disabled it worked again. I think this is related to how yarn hooks the SDK in PnP mode.

I assembled a MCVE over here: https://github-com/BurningEnlightenment/reprex-vscode-ts-filewatcher

<details>
<summary>A/B Experiment Info</summary>
```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
ccp2r3:30993541
dsvsc020:30976470
pythonait:31006305
chatpanelc:31018788
dsvsc021:30996838
9c06g630:31013171
pythoncenvpt:31022790
a69g1124:31038041
dwnewjupytercf:31035177
```
</details>
